packages/@internationalized/date/src/queries.ts

```diff
@@ -79,6 +79,12 @@ export function getDayOfWeek(date: DateValue, locale: string): number {
     dayOfWeek += 7;
   }
 
+  if (firstDayOfWeek) {
+    let offset = (DAY_MAP[firstDayOfWeek] - getWeekStart(locale) + 7) % 7;
+    let diff = (7 + dayOfWeek - offset) % 7;
+    dayOfWeek = diff;
+  } 
```

Comment:
Instead of calculating two different offsets, you could just replace `getWeekStart(locale)` with `DAY_MAP[firstDayOfWeek]` when it is provided?